site/source/cookbook/testing/mocking-imports.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: docs
+category: testing
+title: Test and mock the imports of a widget
+overview: Learn one strategy of mocking imports in a widget
+---
+
+## Objective
+
+This recipe demonstrates one technique of:
+
+1. Testing functionality in your widgets
+2. Mocking imports which your widget relies upon
+
+## Procedure
+
+1. Create your widget:
+
+```js
+const { w } = require('@dojo/widget-core/d');
+const { WidgetBase } = require('@dojo/widget-core/WidgetBase');
+const MyCustomWidget = require('./MyCustomWidget');
+
+class HelloWorld extends WidgetBase {
+    render() {
+        return w(MyCustomWidget, {
+            onclick: () => {
+                this.invalidate();
+            }
+        });
+    }
+}
+
+module.exports = HelloWorld;
+```
+
+2. Add the imports which your test needs. This is the start of a test file which has only one external dependency: `proxyquire`. `proxyquire` allows you to mock/stub JavaScript modules which your widget loads from.
```

Comment:
Add the imports which your test needs -> Import the dependencies for your test

Reply:
Done!